services/repository/init.go

```diff
@@ -71,12 +73,12 @@ func initRepoCommit(ctx context.Context, tmpPath string, repo *repo_model.Reposi
 		defaultBranch = setting.Repository.DefaultBranch
 	}
 
-	if stdout, _, err := gitcmd.NewCommand("push", "origin").
-		AddDynamicArguments("HEAD:" + defaultBranch).
-		WithDir(tmpPath).
-		WithEnv(repo_module.InternalPushingEnvironment(u, repo)).
-		RunStdString(ctx); err != nil {
-		log.Error("Failed to push back to HEAD: Stdout: %s\nError: %v", stdout, err)
+	if err := gitrepo.PushFromLocal(ctx, tmpPath, repo, git.PushOptions{
+		LocalBranch: "HEAD",
```

Comment:
Why it is called `LocalBranch`? Are you sure `HEAD` is a branch name  but not a ref name?